src/go/rpk/pkg/cli/debug/remotebundle/cancel.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package remotebundle
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type cancelResponse struct {
+	Broker   string
+	Canceled bool
+	Error    string
+}
+
+func newCancelCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var noConfirm bool
+	var jobID string
+	cmd := &cobra.Command{
+		Use:   "cancel",
+		Short: "Cancel a remote bundle execution",
+		Long: `Cancel a remote bundle execution.
+
+This command will cancel the debug collection process in a remote cluster that
+you have configured either in your rpk profile, flags or environment variables.
+
+Alternatively, you may provide a --job-id to only cancel the debug bundle with
+the given Job ID.
+
+The flag '--no-confirm' can be used to avoid the confirmation prompt.
```

Comment:
```suggestion
This command cancels the debug collection process in a remote cluster that
you configured in flags, environment variables, or your rpk profile.

Use the flag '--job-id' to only cancel the debug bundle with
the given job ID.

Use the flag '--no-confirm' to avoid the confirmation prompt.
```